data/data/agent/files/usr/local/bin/load-config-iso.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+
+set -e
+
+AGENT_CONFIG_ARCHIVE_FILE="config.gz"
+AGENT_CONFIG_MOUNT="/media/config-image"
+CLUSTER_IMAGE_SET="/etc/assisted/manifests/cluster-image-set.yaml"
+AGENT_CONFIG_MOUNT="/media/config_image"
+
+copy_archive_contents() {
+    tmpdir=$(mktemp --tmpdir -d "config-image--XXXXXXXXXX")
+    cp ${AGENT_CONFIG_MOUNT}/${AGENT_CONFIG_ARCHIVE_FILE} "${tmpdir}"
+    gunzip -f "${tmpdir}"/${AGENT_CONFIG_ARCHIVE_FILE}
+    unzipped_file=$(echo "${tmpdir}/${AGENT_CONFIG_ARCHIVE_FILE}" | cut -d'.' -f1)
+    filelist=$(cpio --list < "${unzipped_file}")
+    echo -e "List of files to copy:\n ${filelist}"
+
+    # Get the releaseImage in the archive and verify it matches the current cluster-image-set
+    release_image=$(grep releaseImage ${CLUSTER_IMAGE_SET} | sed -n -e 's/^.*releaseImage: //p')
+    arch_image_set=$(cpio -icv --to-stdout ${CLUSTER_IMAGE_SET} < "${unzipped_file}")
+    if [[ $(< "${CLUSTER_IMAGE_SET}") != "${arch_image_set}" ]]; then
+       echo "The cluster-image-set in archive does not match current release ${release_image}"
+       cleanup_files
+       return 1
+    fi
+    echo "Archive on ${devname} contains release ${release_image}"
+
+    # Get array from string
+    IFS=',' read -r -a array <<< "${CONFIG_IMAGE_FILES}"
+
+    # Copy expected files from archive, overwriting the existing file
+    for file in "${array[@]}"
+    do
+       if [[ "${file}" = *.* ]]; then
+          cpio -icvdu "${file}" < "${unzipped_file}"
+          if [[ -f ${file} ]]; then
+             echo "Copied ${file}"
+          else
+             echo "Failure copying ${file}"
```

Comment:
q: what's the reason for just logging the copy error? Shouldn't be a blocker?

Reply:
Not necessarily, for example there may not be any files populated in `/etc/assisted/extra-manifests`.
I'll change the text to not log a failure.